src/olympia/addons/models.py

```diff
@@ -846,12 +848,13 @@ def initialize_addon_from_upload(cls, *, data, upload, channel, user):
         guid = data.get('guid')
         old_guid_addon = None
         if guid:  # It's an extension.
-            # Reclaim GUID from deleted add-on.
-            try:
-                old_guid_addon = Addon.unfiltered.get(guid=guid)
-                old_guid_addon.update(guid=None)
-            except ObjectDoesNotExist:
-                pass
+            if waffle.switch_is_active('allow-deleted-guid-reuse'):
+                # Reclaim GUID from deleted add-on.
+                try:
+                    old_guid_addon = Addon.unfiltered.get(guid=guid)
+                    old_guid_addon.update(guid=None)
```

Comment:
Is there follow-up work planned to do something with `AddonGUID`?  It was created really only to handle add-ons reusing guids, which won't be a thing going forward.

Reply:
It has to stay because of existing add-ons, and it's helpful to keep it for new ones as well to have one consistent way to check for guids.